tests/flow/test_variable_length_traversals.py

```diff
@@ -164,3 +164,22 @@ def test09_filtered_edges(self):
         actual_result = redis_graph.query(query)
         expected_result = [['B', 'D']]
         self.env.assertEquals(actual_result.result_set, expected_result)
+
+    # Test traversals with filters on variable-length edges in WITH...OPTIONAL MATCH constructs
+    def test10_filtered_edges_after_segment_change(self):
+        # Test a query that produces the subtree:
+        #   Project
+        #       Filter
+        #           All Node Scan | (a)
+        #   Optional
+        #       Conditional Variable Length Traverse | (a)-[anon_0*1..INF]->(b)
+        #           Argument
+        #
+        # The scan op and the variable-length traversal and its filter are
+        # built in different ExecutionPlan segments. The segments must be
+        # updated before cloning the Optional subtree,
+        # or else the variable-length edge reference will be lost.
```

Comment:
Do we have here a situations where a parent node has children with different AST segments ?
maybe this comment is missing the `Result` operation (the parent of both `Project` and `Optional` ?

Reply:
it is not missing it say subtree so the parent can be Results or any other operation